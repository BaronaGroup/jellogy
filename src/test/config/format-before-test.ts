import {prepareTestKarhu} from '../test-utils'

describe('format-before-test', () => {

  describe('functionality', () => {

    const karhuTest = prepareTestKarhu({
      formatBefore: (logLevel: string, context: string, colorANSI: string, toLog: any[]) => {
        return 'Overridden'
      }
    })

    it('allows replacing the prefix', karhuTest((karhu, output) => {
      karhu.context('test-context').info('Hello')
      expect(output.tracked).toMatchSnapshot()
    }))
  })

  describe('args', () => {
    let args: any

    const karhuTest = prepareTestKarhu({
      formatBefore: (logLevel: string, context: string, colorANSI: string, toLog: any[]) => {
        args = [logLevel, context, colorANSI, toLog]
        return 'Overridden'
      }
    })

    it('is called with the right arguments', karhuTest((karhu, output) => {
      karhu.context('test-context').warn('Hello', 1, {a: 1})
      expect(args).toMatchSnapshot()
    }))
  })
})
