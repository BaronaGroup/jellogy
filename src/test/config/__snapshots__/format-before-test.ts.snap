// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`format-before-test args is called with the right arguments 1`] = `
Array [
  "WARN",
  "test-context",
  "[93m",
  Array [
    "Hello",
    1,
    Object {
      "a": 1,
    },
  ],
]
`;

exports[`format-before-test functionality allows replacing the prefix 1`] = `
Map {
  "NONE" => Array [],
  "DEBUG" => Array [],
  "INFO" => Array [
    Array [
      "Overridden",
      "Hello",
    ],
  ],
  "WARN" => Array [],
  "ERROR" => Array [],
}
`;
